### DISCUSSION TITLE
How can I use a comma inside a regex list?

### DISCUSSION THREAD

**Discussion by Machine-Maker** [checkstyle/checkstyle#16082] [Q&A]
It seems to split on commas and I'm trying to include it. Escaping with \ didn't seem to work either.

```xml
<module name="IllegalImport">
  <property name="regexp" value="true"/>
  <property name="illegalClasses" value="javax\.annotation\.(Nullable|Nonnull)"/>
  <property name="illegalClasses" value="^[^\.A-Z]+(\.[^\.A-Z]+)+?(\.[A-Z][^\.]*){2,}$"/>
</module>
```
Specifically the `{2,}` part keeps breaking the expression.

----

In this specific example, I could just duplicate that capture group, but I'm wondering if there is a general solution in case this comes up in the future.

**Comment by romani** [comment]
Somewhere in issues I pretty sure we decided to sacrifice `,` in regexp, as almost never used symbol. 
And we forgot about this regexp syntax :( .

There is no way. Sorry.

**Comment by rnveach** [comment]
`illegalClasses` is a string array first, then a regular expression second (if regexp is enabled).

https://checkstyle.org/property_types.html#String.5B.5D
String arrays on broken on the commas first, so this is why your expression isn't working as it is not seen as a single regular expression.
